### PR TITLE
Change loss API

### PR DESCRIPTION
**Pull request by chengxuz**
Related to issue #34 . I also added some instructions about how to use it. 
I think we might need to put those instructions to some markdown documents on github.

### CODE REVIEW

File: tfutils/utils.py [hunk 1]
```diff
@@ -173,18 +173,48 @@ def get_loss(inputs,
...
              loss_per_case_func,
+             loss_per_case_func_params={'_outputs': 'logits', '_targets_$all': 'labels'},
              agg_func=None,
              loss_func_kwargs=None,
              agg_func_kwargs=None):
     if loss_func_kwargs is None:
         loss_func_kwargs = {}
     if not isinstance(targets, (list, tuple, np.ndarray)):
         targets = [targets]
+    targets = list(targets)
```
Comment (qbilius) [inline tfutils/utils.py]: Seems like this could have an unintended consequence of producing something like [np.array([...])].

File: tfutils/utils.py [hunk 2]
```diff
@@ -173,18 +173,48 @@ def get_loss(inputs,
...
     if len(targets) == 1:
         labels = inputs[targets[0]]
     else:
         labels = [inputs[t] for t in targets]
-    loss = loss_per_case_func(logits=outputs, labels=labels, **loss_func_kwargs)
+
+    flag_without = True
```
Comment (qbilius) [inline tfutils/utils.py]: Would be easier to understand if called `flag_with_out` and `flag_with_tar`

File: tfutils/utils.py [hunk 3]
```diff
@@ -173,18 +173,48 @@ def get_loss(inputs,
...
+        labels = inputs[targets[0]]
+    else:
+        labels = [inputs[t] for t in targets]
+
+    if not flag_withtar:
+        labels = []
+    if flag_without:
+        labels.insert(0, outputs)
+    loss = loss_per_case_func(*labels, **loss_func_kwargs)
+
```
Comment (qbilius) [inline tfutils/utils.py]: I'm a bit confused why there is `labels` argument. Just put outputs and labels into `**loss_func_kwargs`!

File: tfutils/utils.py (hunk 1 above)
Comment (chengxuz) [inline tfutils/utils.py]: I was thinking that targets would be list of strings. So I actually don't understand why it could be np.ndarray. But even if it could be, list(np.ndarray) is still well-defined.

File: tfutils/utils.py (hunk 3 above)
Comment (chengxuz) [inline tfutils/utils.py]: This is for some situation where the users actually want positional parameters, for whatever reason. For example,  if the loss_per_case_func_params is empty, then the API will be the same as before.

File: tfutils/utils.py (hunk 2 above)
Comment (chengxuz) [inline tfutils/utils.py]: I see. Will do that.

File: tfutils/utils.py (hunk 1 above)
Comment (qbilius) [inline tfutils/utils.py]: Suppose somebody passed targets like `np.array(['labels', 'labels2'])`. Then this will not work.

File: tfutils/utils.py (hunk 3 above)
Comment (qbilius) [inline tfutils/utils.py]: OK, I guess it's ok then.

File: tfutils/utils.py (hunk 1 above)
Comment (chengxuz) [inline tfutils/utils.py]: list() can take np.array input. See below:

```
In [2]: t = np.array(['labels', 'labels2'])

In [3]: list(t)
Out[3]: ['labels', 'labels2']

```

Comment (qbilius) [inline tfutils/utils.py]: OK, I take it back. I tested in incorrectly before commenting (did `[np.array(['labels', 'labels2'])]`)
